pandas/core/common.py

```diff
@@ -782,6 +782,13 @@ def take_nd(arr, indexer, axis=0, out=None, fill_value=np.nan,
         will be done.  This short-circuits computation of a mask.  Result is
         undefined if allow_fill == False and -1 is present in indexer.
     """
+
+    if is_categorical(arr):
+        result = take_nd(arr.get_values(), indexer, axis=axis, out=out,
+                         fill_value=fill_value, mask_info=mask_info,
```

Comment:
no, you send this from a higher level (e.g. arr is actually a Series/CategoricalIndex object when you start).

You must use the ``take_1d`` in Categorical, and NOT create one here, this is moving too specific knowledge to this function

something like:

```
if is_categorical_dtype(arr):
    arr = getattr(arr,'value',arr)
    return arr.take_1d(....)

...
```